doc/users/explain/figures.rst

```diff
@@ -0,0 +1,187 @@
+``matplotlib.use('Qt5Agg')`` at the beginning of a session or script.
+
+When run from a script, or interactively (e.g. from an
+`iPython shell <https://https://ipython.readthedocs.io/en/stable/>`_) the Figure
+will not be shown until we call ``plt.show()``. The Figure will appear in
+a new GUI window, and usually will have a toolbar with Zoom, Pan, and other tools
+for interacting with the Figure.  By default, ``plt.show()`` blocks
+further interaction from the script or shell until the Figure window is closed,
+though that can be toggled off for some pusposes.
+
+Note that if you are on a client that does not have access to a windowing
+system, the Figure will fallback to being drawn using the "Agg" backend, and
+cannot be viewed, though it can be :ref:`saved <saving_figures>`.
+
+.. _creating_figures:
+
+Creating Figures
+================
+
+By far the most common way to create a figure is using the
+:doc:`pyplot </tutorials/introductory/pyplot>` interface. As noted in
+:ref:`api_interfaces`, the pyplot interface serves two purposes.  One is to spin
+up the Backend and keep track of GUI windows. The other is a global state for
+Axes and Artists that allow a short form API to plotting methods. In the
+example above, we only use pyplot for the first purpose, to create the Figure
+object, ``fig``.
+
+In addition to `~.pyplot.figure`, there are convenience methods to create Axes
+at the same time that the figure is created since these two actions are so commonly
+done at the same time. A simple grid can be achieved with `~.pyplot.subplots` (which
+simply wraps `~.Figure.subplots`):
+
+.. plot::
+    :include-source:
+
+    fig, axs = plt.subplots(2, 2, figsize=(4, 3), layout='constrained')
+
+More complex grids can be achieved with `~.pyplot.subplot_mosaic` (which wraps
+`~.Figure.subplot_mosaic`):
+
+.. plot::
+    :include-source:
+
+    fig, axs = plt.subplot_mosaic([['A', 'right'], ['B', 'right']],
+                                  figsize=(4, 3), layout='constrained')
+    for ax_name in axs:
+        axs[ax_name].text(0.5, 0.5, ax_name, ha='center', va='center')
+
+Sometimes we want to have a nested layout in a Figure, with two or more sets of
+Axes that do not necessarily share the same subplot grid.
+We can use `~.Figure.add_subfigure` or `~.Figure.subfigures` to create virtual
+figures inside a parent Figure; see
+:doc:`/gallery/subplots_axes_and_figures/subfigures` for more details.
+
+.. plot::
+    :include-source:
+
+    fig = plt.figure(layout='constrained', facecolor='beige')
+    fig.suptitle('Figure')
+    figL, figR = fig.subfigures(1, 2)
+    figL.set_facecolor('0.9')
+    axL = figL.subplots(2, 1, sharex=True)
+    figL.suptitle('Left subfigure')
+    figR.set_facecolor('0.7')
+    axR = figR.subplots(1, 2, sharey=True)
+    figR.suptitle('Right subfigure')
+
+It is possible to directly instantiate a `.Figure` instance.  This is usually
+only necessary if you want to create your own GUI application and not use
```

Comment:
I think we should include a note that this is how to opt-out of the global registry of `Figure` s that `pyplot` does and is a good choice for batch or webserver applications where a given process is expected to create many `Figure` objects through its life.

Reply:
I didn't want to get into a lot of use cases here - presumably folks who need this are sophisticated enough to not need this document at all.  Here is what I changed it to:

```
This is usually only necessary if you want to create your
own GUI application or service that you do not want carrying the pyplot global
state.
```